### DISCUSSION TITLE
Entering paswords interactively

### DISCUSSION THREAD

**Discussion by MihaMarkic** [cake-build/cake#3661] [Q&A] [answered]
Is there a prompt that lets one type the password while not showing it in console?
I know I could implement it myself using Console.ReadKey though.

**Comment by pascalberger** [comment, marked as answer]
Cake ships with Spectre.Console library which supports [secrets text prompt](https://spectreconsole.net/prompts/text#secrets):

```
var password = AnsiConsole.Prompt(
    new TextPrompt<string>("Enter [green]password[/]")
        .PromptStyle("red")
        .Secret());
``` 

See [Documentation](https://cakebuild.net/docs/writing-builds/console-output#advanced-output) for details.
